### PR TITLE
Remove hydrate option from createRoot

### PR DESCRIPTION
**Pull request by salazarm**
## Summary

Removing `hydrate: true` and `hydrateOptions` from `createRoot`. Instead `hydrateRoot` should be used.

## How did you test this change?

Updated jest tests. There is one `useMutableSourceHydration` test that I'm unsure about. Added a comment above it.

### CODE REVIEW

File: packages/react-reconciler/src/__tests__/useMutableSourceHydration-test.js [hunk 1]
```diff
@@ -438,19 +408,17 @@ describe('useMutableSourceHydration', () => {
...
-      ],
-      {withoutStack: 1},
+      'Warning: Text content did not match. Server: "1" Client: "2"',
     );
-    expect(Scheduler).toHaveYielded([2, 'a:two']);
     expect(source.listenerCount).toBe(1);
+    if (gate(flags => flags.enableSyncDefaultUpdates)) {
+      expect(Scheduler).toHaveYielded([2, 'a:two']);
+    } else {
+      // Not sure about this...
```
Comment (salazarm) [inline packages/react-reconciler/src/__tests__/useMutableSourceHydration-test.js:420]: @acdlite  Unsure why I needed to do this. Could you enlighten me please? :)

Comment (acdlite) [reply packages/react-reconciler/src/__tests__/useMutableSourceHydration-test.js:420]: `hydrateRoot` is synchronous whereas `root.render` is not

**Review by acdlite — APPROVED** [review]

**Comment by sebmarkbage** [comment]
We should probably keep a warning if someone passes it since it was around in early versions a long time.

**Comment by sebmarkbage** [comment]
Nice. It's like taking a shower. Feels fresh.

**Pull request merged**